testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -434,23 +434,24 @@
   )
 end
 
-When(/^I enter the hostname of "([^"]*)" as "([^"]*)"$/) do |host, hostname|
+When(/^I enter the hostname of "([^"]*)" as "([^"]*)"$/) do |host, field|
   system_name = get_system_name(host)
   log "The hostname of #{host} is #{system_name}"
-  step %(I enter "#{system_name}" as "#{hostname}")
-end
-
-When(/^I select the hostname of "([^"]*)" from "([^"]*)"$/) do |host, hostname|
-  case host
-  when 'proxy'
-    # don't select anything if not in the list
-    next if $proxy.nil?
-    step %(I select "#{$proxy.full_hostname}" from "#{hostname}")
-  when 'sle_minion'
-    step %(I select "#{$minion.full_hostname}" from "#{hostname}")
-  when 'build_host'
-    step %(I select "#{$build_host.full_hostname}" from "#{hostname}")
+  step %(I enter "#{system_name}" as "#{field}")
+end
+
+When(/^I select the hostname of "([^"]*)" from "([^"]*)"((?: if present)?)$/) do |host, field, if_present|
+  if if_present.empty?
+    node = get_target(host)
+  else
+    begin
+      node = get_target(host)
+    rescue
+      log "#{host} is not deployed, not trying to select it"
+      return
+    end
   end
+  step %(I select "#{node.full_hostname}" from "#{field}")
 end
```

Comment:
I think this step must be re-think, the algorithm is not readable to me.
I don't have time now to think about a proposal, let me some days, and we can re-discuss it.

Reply:
if there's a way to continue normally from the `rescue` block, then that would be more readable, yes

another way would be not to use `get_target` but i'm not that warm for that because `get_target` is what we have everywhere.

or any other solution you find.

Reply:
ah, you can exit a `rescue` simply by doing nothing. Is it better now?

Reply:
Is it better now?